### PR TITLE
fix(engine): ExecutableActivity.getFlowScope() may produce NPE

### LINKED ISSUE #9400: NPE: Cannot invoke `ExecutableActivity.getFlowScope()` because `innerActivity` is null
**Describe the bug**

While deploying a BPMN resource, transforming a multi-instance activity fails with a NPE.

The deployment is rejected with the following rejection message:

```
Command 'CREATE' rejected with code 'INVALID_ARGUMENT': Expected to deploy new resources, but encountered the following errors:
'diagram_1.bpmn': Cannot invoke "io.camunda.zeebe.engine.processing.deployment.model.element.ExecutableActivity.getFlowScope()" because "innerActivity" is null
```

The following process causes the issue:

image

But, I'm able to deploy this process:

image

The only difference between the processes is multi-instance activity. If the multi-instance activity is in the first pool (i.e. the first process in the XML) then I can deploy the process successfully. Otherwise, the deployment fails. 

**To Reproduce**

1. deploy the process: [issue-9400.bpmn](https://github.com/camunda/zeebe/files/9562679/issue-9400.bpmn.txt) (with the correct file extension `.bpmn`)
2. verify that the deployment is rejected

**Expected behavior**

I can deploy a process that contains multiple pools with a multi-instance activity.

I can execute and complete an instance of this process.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
2022-09-14 04:09:21.842 ERROR 1 --- [   -zb-actors-2] io.camunda.zeebe.broker.process          : Unexpected error while processing resource 'carPolicyRequestProcess-init.bpmn'

java.lang.NullPointerException: Cannot invoke "io.camunda.zeebe.engine.processing.deployment.model.element.ExecutableActivity.getFlowScope()" because "innerActivity" is null
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.MultiInstanceActivityTransformer.transform(MultiInstanceActivityTransformer.java:53) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.MultiInstanceActivityTransformer.transform(MultiInstanceActivityTransformer.java:29) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.deployment.model.transformation.TransformationVisitor.visit(TransformationVisitor.java:42) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.model.bpmn.traversal.TypeHierarchyVisitor.visit(TypeHierarchyVisitor.java:39) ~[zeebe-bpmn-model-8.0.5.jar:8.0.5]
	at io.camunda.zeebe.model.bpmn.traversal.ModelWalker.walk(ModelWalker.java:61) ~[zeebe-bpmn-model-8.0.5.jar:8.0.5]
	at io.camunda.zeebe.engine.processing.deployment.model.transformation.BpmnTransformer.transformDefinitions(BpmnTransformer.java:117) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.deployment.transform.BpmnResourceTransformer.transformResource(BpmnResourceTransformer.java:69) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.deployment.transform.DeploymentTransformer.transformResource(DeploymentTransformer.java:120) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.deployment.transform.DeploymentTransformer.transform(DeploymentTransformer.java:97) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.deployment.DeploymentCreateProcessor.processRecord(DeploymentCreateProcessor.java:100) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor.processRecord(TypedRecordProcessor.java:54) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.lambda$processInTransaction$3(ProcessingStateMachine.java:300) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at org.camunda.community.eze.db.EzeDbTransaction.run(EzeDbTransaction.kt:34) ~[eze-1.0.2.jar:1.0.2]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processInTransaction(ProcessingStateMachine.java:290) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processCommand(ProcessingStateMachine.java:253) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.tryToReadNextRecord(ProcessingStateMachine.java:213) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.readNextRecord(ProcessingStateMachine.java:189) ~[zeebe-workflow-engine-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.util.sched.ActorJob.invoke(ActorJob.java:79) ~[zeebe-util-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.util.sched.ActorJob.execute(ActorJob.java:44) ~[zeebe-util-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.util.sched.ActorTask.execute(ActorTask.java:122) ~[zeebe-util-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:97) ~[zeebe-util-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.util.sched.ActorThread.doWork(ActorThread.java:80) ~[zeebe-util-8.0.4.jar:8.0.4]
	at io.camunda.zeebe.util.sched.ActorThread.run(ActorThread.java:189) ~[zeebe-util-8.0.4.jar:8.0.4]
```

</p>
</details>

Archived Logs: https://drive.google.com/file/d/1VarHjCc-XQZ6bZz7PUE4MmdDy0pQ5Xt9/view?usp=sharing

**Environment:**

- Zeebe Version: 8.0.1

### PR DESCRIPTION
**Pull request by skayliu**
## Description

Fix `ExecutableActivity.getFlowScope()` may produce `NullPointerException`.

## Related issues

closes #9400

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by saig0** [comment]
@skayliu thank you for your contribution. :tada: We'll have a look at your PR in the next few days. :eyes:

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
Nice work root-causing this one @skayliu! 🚀  Your fix would work but I would propose a different solution. Please have a look 👀

File: engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/CollaborationDeploymentTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.mockito.verification.VerificationWithTimeout;
+
+@RunWith(Parameterized.class)
+public final class CollaborationDeploymentTest {
```
Comment (remcowesterhoud) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/CollaborationDeploymentTest.java:37]: 👍 I like you regression test

File: engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/CollaborationDeploymentTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  public Function<DeploymentClient, Record<DeploymentRecordValue>> performDeployment;
+
+  @Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(
+        new Object[] {"collaboration with NPE error one", deploy("/processes/issue-9400-NPE.bpmn")},
+        new Object[] {
+          "collaboration with NPE error two",
+          deploy("/processes/issue-9400-carPolicyRequestProcess.bpmn")
+        },
```
Comment (remcowesterhoud) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/CollaborationDeploymentTest.java:61]: 🔧 I think we could remove this specific scenario. The other 2 processes already reproduce the problem. There is no need for the complex process. If the test fails it only provides confusing when debugging.

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
Thanks @skayliu! I have one more small comment which I missed in the previous review. Please let me know what you think.

File: engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/CollaborationDeploymentTest.java (hunk 2 above)
Comment (skayliu) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/CollaborationDeploymentTest.java]: Done.

**Review by remcowesterhoud — APPROVED** [review]
Thanks for the fix! 🍪 

bors merge

**Pull request merged**